dapps/digital-twin-detail/src/components/sharing/SharingOverview.ts

```diff
@@ -0,0 +1,234 @@
+  containers: Array<string>;
+  name: Promise<string>;
+  type: Promise<string>;
+}
+
+/**
+ * Format of resolved "search" results
+ */
+interface SharedUserInterface {
+  address: string; // account address
+  containerNames: Array<string>; // list of containers, mapped to it's names
+  containers: Array<string>; // list of containers, for that the user is permitted;
+  icon: string; // account type material icon
+  name: string; // users name, alias or account address
+  type: string; // account type
+}
+
+interface ColumnInterface {
+  key: string;
+  label: string;
+  sortable?: boolean;
+  tdClass?: string;
+  thClass?: string;
+}
+
+@Component({ })
+export default class SharingOverview extends mixins(EvanComponent) {
+  columns: ColumnInterface[ ] = [];
+
+  data: SharedUserInterface[ ] = [];
+
+  isActiveSearch = false;
+
+  loading = true;
+
+  reverse = false;
+
+  selectedFilter = 'all';
+
+  searchTerm = '';
+
+  sortBy = 'name';
+
+  filters = [
+    { type: 'unshared', icon: 'help-circle-outline' },
+    { type: 'user', icon: 'account-outline' },
+    { type: 'company', icon: 'office-building' },
+    { type: 'all', icon: 'share-variant' },
+  ];
+
+  /**
+   * Resolve permissions for the twin and containers.
+   */
+  async created(): Promise<void> {
+    this.columns = [
+      {
+        key: 'icon',
+        label: '',
+        thClass: 'th-icon',
+      },
+      {
+        key: 'name',
+        label: this.$t('_twin-detail.sharings.table.name'),
+        sortable: true,
+        tdClass: 'truncate',
+      },
+      {
+        key: 'containers',
+        label: this.$t('_twin-detail.sharings.table.containers'),
+        tdClass: 'truncate',
+      },
+    ];
+
+    await this.resolvePermissions();
+  }
+
+  /**
+   * Load all permissions for the current opened twins.
+   */
+  async resolvePermissions(): Promise<void> {
+    const runtime = this.getRuntime();
+    const { twin: { containerAddresses } }: { twin: DAppTwin } = this.$store.state;
+    const permittedUsers: { [address: string]: LoadingSharedUserInterface } = { };
+
+    // iterate through all containers and resolve the permissions
+    await Promise.all(containerAddresses.map(async (containerAddress: string): Promise<void> => {
+      /* Hint: Container API invites every shared user generally into the member role. As a result
+       * of this, we can simply resolve all members from the rights and roles, to check with which
+       * users anything was shared. */
+      const contract = runtime.contractLoader.loadContract('DataContract', containerAddress);
+      const roleMap = await runtime.rightsAndRoles.getMembers(contract);
+      const unique = Array.from(new Set([].concat(...Object.values(roleMap))));
+
+      // iterate through all permitted users and apply them to the permitted users mapping
+      unique.forEach((address: string): void => {
+        if (address !== runtime.activeAccount) {
+          if (!permittedUsers[address]) {
```

Comment:
isn't it already unique and the check should be obsolete?

Reply:
No, it could be not unique over multiple containers.